app/tut/ga/creatlach.md

```diff
@@ -0,0 +1,967 @@
+[[míshlachtmhar|messy]]. Conas a scríobhaimid an ríomhchlár sin
+[[i slí níos deise|in a nicer way]]?
+
+Anois féachfaimid ar iompraíocht na hoibiachta agus an liosta gníomhartha is féidir linn a scríobh
+sa chreatlach.
+
+Ar dtús féachfaimid ar an ngníomh `tarraing`. Is féidir linn an gníomh sin a bhogadh isteach sa
+chreatlach `Ciorcal`:
+
+```{.setanta .numberLines}
+>-- Creatlach folamh.
+creatlach Ciorcal {
+
+    >-- Tarraing an ciorcal.
+    gníomh tarraing(ciorc) {
+        >-- Athraigh dath an phinn.
+        dath@stáitse(dath@ciorc)
+
+        >-- Tarraing an ciorcal leis an x comhordanáid, y comhordanáid agus an ga.
+        ciorcal@stáitse(x@ciorc, y@ciorc, ga@ciorc)
+    }
+
+}
+```
+
+Anois nuair a cruthaímid oibiacht ón gcreatlach `Ciorcal`, beidh an gníomh `tarraing` ina bhall. Mar
+shampla, ba fhéidir linn é a roghnú mar seo:
+
+```{.setanta .numberLines}
+c := déan_ciorcal(100, 100, 50, "dearg")
+
+tarraing@c
+```
+
+Ansin is féidir linn glaoch air mar a rinneamar roimhe:
+
+```{.setanta .numberLines}
+c := déan_ciorcal(100, 100, 50, "dearg")
+
+tarraing@c(c)
+```
+
+Oibríonn an cód seo, bain triail as:
+
+{{{s
+>-- Creatlach folamh.
+creatlach Ciorcal {
+    >-- Tarraing an ciorcal.
+    gníomh tarraing(ciorc) {
+        >-- Athraigh dath an phinn.
+        dath@stáitse(dath@ciorc)
+
+        >-- Tarraing an ciorcal leis an x comhordanáid, y comhordanáid agus an ga.
+        ciorcal@stáitse(x@ciorc, y@ciorc, ga@ciorc)
+    }
+}
+
+gníomh déan_ciorcal(x, y, ga, dath) {
+    >-- Cruthaigh ciorcal folamh.
+    c := Ciorcal()
+
+    >-- Cuir na luachanna san oibiacht.
+    x@c = x
+    y@c = y
+    ga@c = ga
+    dath@c = dath
+
+    >-- Tabhair an ciorcal ar ais.
+    toradh c
+}
+
+>-- Ríomh an t-achar. (achar = pi * ga^2).
+gníomh achar(ciorc) {
+    toradh pi@mata * ga@ciorc * ga@ciorc
+}
+
+>-- Ríomh an t-imlíne. (imlíne = 2 * pi * ga).
+gníomh imlíne(ciorc) {
+    toradh 2 * pi@mata * ga@ciorc
+}
+
+c := déan_ciorcal(100, 100, 50, "dearg")
+tarraing@c(c)
+scríobh("Achar: ", achar(c))
+scríobh("Imlíne: ", imlíne(c))
+}}}
+
+[[Faraor|Alas]], tá an líne seo [[an ait|very strange]]: `tarraing@c(c)`. Bhí orainn `c` a úsáid
+faoi dhó. An bhfuil bealach níos fearr ann? Tá! Is féidir linn an [[eochairfhocail|keyword]]
+[[`seo`{.setanta}|this]] a úsáid chun cód níos simplí a chruthú.
+
+### Seo
+
+Nuair a bhaineann tú úsáid as creatlach chun oibiacht a chruthú, tugaimid **"[[ásc|instance]]"** ar
+an oibiacht sin. Is féidir linn a lán áisc a chruthú ó gcreatlach amháin. Nuair a roghnaíonn tú
+gníomh ó ásc éigin (m.sh. `tarraing@c`{.setanta}), tá an gníomh sin [[nasctha|connected]] leis an
+ásc sin. Is féidir linn an eochairfhocail `seo`{.setanta} a úsáid chun sochar a bhaint as an nasc
+sin.
+
+Nuair a cruthaímid gníomh i gcreatlach, (díreach mar a rinneamar le `tarraing` thuas) is féidir linn
+`seo`{.setanta} chun [[tagairt|reference]] a dhéanamh don ásc atá nasctha leis an gníomh nuair a
```

Comment:
is féidir linn tagairt don ásc atá nasctha leis an gníomh tríd seo{.setanta} a úsáid

Reply:
Athraithe